packages/is-shallow-equal/arrays/package.json

```diff
@@ -0,0 +1,3 @@
+{
```

Comment:
I suppose this is needed to support usage of `@wordpress/is-shallow-equal/arrays` ?

Does it work if this were `/index.js` of `module.exports = require( '../lib/arrays' );` ? Or does TypeScript take issue with that?

The reason I ask is two-fold:

- AFAIK, a `package.json` requires at minimum `name` and `version` fields to be valid [[1]](https://nodejs.org/en/knowledge/getting-started/npm/what-is-the-file-package-json/) [[2]](https://docs.npmjs.com/files/package.json). Whether the latter is relevant only for published packages, I'm neither sure nor feel confident one way or the other.
- It would be nice if we could avoid the `.npmpackagejsonlintrc.json`

Reply:
I pushed a change with it. It seems to work fine with node and I'd expect with bundlers as well, however TypeScript (and IDE support) do not see to be able to find the types for imports like `@wordpress/is-shallow-equals/arrays`. I don't know if extra package.json with `types` fields will help, but I'm inclined not to provide first class support for this feature.

This is something that could be revisited, I'm curious about the motivations for allowing/supporting this.

Reply:
> This is something that could be revisited, I'm curious about the motivations for allowing/supporting this.

Was there anywhere you saw that we were referencing it this way? I vaguely recall something about `@wordpress/is-shallow-equal/array`, but at least in [how it's documented currently](https://github.com/WordPress/gutenberg/tree/master/packages/is-shallow-equal#usage), the object/array-specific functions are available as named exports. They're also available on the `wp` global as `wp.isShallowEqual.isShallowEqualArrays`, `wp.isShallowEqual.isShallowEqualObjects`. So I'm not sure it's something we considered as part of the public API, and was structured as such more to the point of general file structure?

Reply:
Sounds like we can drop support for the direct imports via `…/arrays` and `…/objects`. That will help simplify things a little bit 👍

Reply:
a3598cc removes the additional files to support direct access via `@wordpress/is-shallow-equal/{array,object}`.